corehq/apps/domain/decorators.py

```diff
@@ -58,7 +58,16 @@ def login_and_domain_required(view_func):
     @wraps(view_func)
     def _inner(req, domain, *args, **kwargs):
         user = req.user
-        domain_name, domain = load_domain(req, domain)
+        domain_identifier = domain
+        domain_name, domain = load_domain(req, domain_identifier)
+        if not domain:
+            domain = Domain.get_by_alias(domain_identifier)
+            if domain:
+                domain_name, domain = load_domain(req, domain.name)
+                return HttpResponseRedirect(
+                    req.path.replace('/' + domain_identifier + '/', '/' + domain_name + '/')
```

Comment:
can it least be `/a/domain/` instead of `/domain/`? assume there wasn't a better way to do this.

also small but the more pythonic way to handle the string concats would be `'/{}/'.format(domain)`